### ISSUE TITLE
Allow passing a buffer into `uuid.parse`?

### ISSUE THREAD

**Issue by ctavan** [uuidjs/uuid#497]
## Is your feature request related to a problem? Please describe.

Currently our `vX` methods allow passing an input buffer and an offset as second/third parameter. The `parse` method on the other hand always creates and returns a `new Uint8Array(16)`.

This adds a slight inconsistency since for the buffer cases so far it was always possible to define & hand in the buffer on your own.

## Describe the solution you'd like

Would it make any sense to allow passing an input buffer/offset here as well?

## Describe alternatives you've considered

Wait until somebody wants this API, only then extend the API.

What are your thoughts @broofa?

**Comment by broofa** [comment]
Agreed.  Wait until someone complains.

Note: The ability for users to pass buffers and offsets into these methods was something I added early-on when I was obsessing over performance.  But as I've mentioned, I'm less concerned with performance and I'm of the opinion that may have been a mistake.  It unnecessarily complicates the UI for what I suspect is a very narrow edge case.

tl;dr: I actually think it'd be better to remove the `buffer` and `offset` options altogether.  But, failing that, at least being consistent in their support would be good.

**Comment by ctavan** [comment]
OK, then let's close this until it is brought up again.

I also tend to agree w.r.t. removing the `buffer` parameter, but I think it would make sense to analyze real-world usage of this library (e.g. similar to [the analysis I did in the uuid-proposal](https://github.com/tc39/proposal-uuid/tree/master/analysis)).
